DuckDuckGo/DefaultNetworkProtectionVisibility.swift

```diff
@@ -0,0 +1,97 @@
+import NetworkProtection
+import Core
+
+struct DefaultNetworkProtectionVisibility: NetworkProtectionFeatureVisibility {
+    private let privacyConfigurationManager: PrivacyConfigurationManaging
+    private let networkProtectionTokenStore: NetworkProtectionTokenStore?
+    private let networkProtectionAccessManager: NetworkProtectionAccess?
+    private let featureFlagger: FeatureFlagger
+    private let userDefaults: UserDefaults
+
+    init(privacyConfigurationManager: PrivacyConfigurationManaging = ContentBlocking.shared.privacyConfigurationManager,
+         networkProtectionTokenStore: NetworkProtectionTokenStore? = NetworkProtectionKeychainTokenStore(),
+         networkProtectionAccessManager: NetworkProtectionAccess? = NetworkProtectionAccessController(),
+         featureFlagger: FeatureFlagger = AppDependencyProvider.shared.featureFlagger,
+         userDefaults: UserDefaults = .networkProtectionGroupDefaults) {
+        self.privacyConfigurationManager = privacyConfigurationManager
+        self.networkProtectionTokenStore = networkProtectionTokenStore
+        self.networkProtectionAccessManager = networkProtectionAccessManager
+        self.featureFlagger = featureFlagger
+        self.userDefaults = userDefaults
+    }
+
+    /// A lite version with fewer dependencies
+    /// We need this to run shouldMonitorEntitlement() check inside the token store
+    static func forTokenStore() -> DefaultNetworkProtectionVisibility {
+        DefaultNetworkProtectionVisibility(networkProtectionTokenStore: nil, networkProtectionAccessManager: nil)
+    }
+
+    func isWaitlistBetaActive() -> Bool {
+        privacyConfigurationManager.privacyConfig.isSubfeatureEnabled(NetworkProtectionSubfeature.waitlistBetaActive)
+    }
+    
+    func isWaitlistUser() -> Bool {
+        guard let networkProtectionTokenStore, let networkProtectionAccessManager else {
+            preconditionFailure("networkProtectionTokenStore and networkProtectionAccessManager must be non-nil")
+        }
+
+        let hasLegacyAuthToken = {
+            guard let authToken = try? networkProtectionTokenStore.fetchToken(),
+                  !authToken.hasPrefix(NetworkProtectionKeychainTokenStore.authTokenPrefix) else {
+                return false
+            }
+            return true
+        }()
+        let hasBeenInvited = {
+            let vpnAccessType = networkProtectionAccessManager.networkProtectionAccessType()
+            return vpnAccessType == .waitlistInvited || vpnAccessType == .inviteCodeInvited
+        }()
+
+        return hasLegacyAuthToken || hasBeenInvited
+    }
+
+    // todo - https://app.asana.com/0/0/1206844038943626/f
+    func isPrivacyProLaunched() -> Bool {
+        if let subscriptionOverrideEnabled = userDefaults.subscriptionOverrideEnabled {
+#if ALPHA || DEBUG
+            // only allows disabling subscription in Alpha or Debug build
+            return subscriptionOverrideEnabled
+#else
+            return true
+#endif
+        }
```

Comment:
This should make it easier to simulate the PP launch. Go to the Debug menu to update the override.

Reply:
Nice :+1: